cartographer/mapping_2d/sparse_pose_graph.cc

```diff
@@ -322,6 +325,15 @@ common::Time SparsePoseGraph::GetLatestScanTime(
   return time;
 }
 
+void SparsePoseGraph::SetInitialTrajectoryPose(const int trajectory_id,
+                                               const transform::Rigid3d& pose,
+                                               const common::Time& time) {
+  common::MutexLocker locker(&mutex_);
+  pose_initialization_map[trajectory_id] = pose;
+  trajectory_connectivity_state_.Add(trajectory_id);
+  trajectory_connectivity_state_.Connect(0, trajectory_id, time);
```

Comment:
Here actually assumes that `Trajectory 0` exist. Should we skip trajectory_connectivity part if the given trajectory is 0?

Reply:
Does setting an initial pose for Trajectory 0 make sense? Isn't the global coordinate system actually defined to be the origin of trajectory 0?

Reply:
It is not necessarily the same. It happened to the same because trajectories always start at (0,0,0).

Reply:
I think throwing an error in that case might also be a viable option. 
@ojura the start of trajectory 0 is so far fixed to zero at least, yes. Also in #254 it was discussed and decided to keep it. I personally don't care, but since the interface for starting a trajectory would change anyway and the alternative option would be to throw an error, we could reconsider this option.